sktime/clustering/include/metric.h

```diff
@@ -16,17 +16,17 @@ namespace py = pybind11;
 
 class Metric {
 public:
+    virtual ~Metric() = default;
     virtual double compute_d(const double* xs, const double* ys, std::size_t dim) const = 0;
     virtual float compute_f(const float* xs, const float* ys, std::size_t dim) const = 0;
 
     template<typename T>
     T compute(const T* xs, const T* ys, std::size_t dim) const {
-        if constexpr (std::is_same_v<T, float>) {
-            return compute_f(xs, ys, dim);
-        } else {
-            return compute_d(xs, ys, dim);
-        }
+        return compute_d(xs, ys, dim);
     }
+	float compute(const float* xs, const float* ys, std::size_t dim) const {
+		return compute_f(xs, ys, dim);
+	}
```

Comment:
this isnt actually doing the same thing as far as i can tell, the way to work around `if constexpr` would be with `std::enable_if`